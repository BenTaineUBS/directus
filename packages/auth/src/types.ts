export interface User {
	id: string;
	first_name: string | null;
	last_name: string | null;
	email: string | null;
	password: string | null;
	status: 'active' | 'suspended';
	role: string | null;
	provider: string;
	provider_key: string | null;
	provider_data: string | null;
}

export interface AuthProviderConfig {
	driver: string;
	config: any;
}

export interface AuthManagerProviderConfig {
	[key: string]: AuthProviderConfig;
}

export interface AuthManagerConfig {
	default?: string;
	providers?: AuthManagerProviderConfig;
}
