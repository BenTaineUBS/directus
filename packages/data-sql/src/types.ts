import type { AbstractQueryNodeSortTargets } from '@directus/data';

interface SqlStatementColumn {
	type: 'primitive';
	table: string;
	column: string;
}
export interface SqlStatementSelectColumn extends SqlStatementColumn {
	as?: string;
}

// export interface SqlStatementSelectFn {
// 	type: 'fn';
// 	fn: string;
// 	args: (string | number | boolean)[];
// 	table: string;
// 	column: string;
// 	as?: string;
// }

// export interface SqlStatementSelectJson {
// 	type: 'json';
// 	table: string;
// 	column: string;
// 	as?: string;
// 	path: string;
// }

/**
 * Used for parameterized queries.
 */
type ParameterIndex = {
	/** Indicates where the actual value is stored in the parameter array */
	parameterIndex: number;
};

/**
 * This is an abstract SQL query which can be passen to all SQL drivers.
 *
 * @example
 * ```ts
 * const query: SqlStatement = {
 *  select: [id],
 *  from: 'articles',
 *  limit: 0,
 * 	parameters: [25],
 * };
 * ```
 */
export interface AbstractSqlQuery {
	select: SqlStatementSelectColumn[];
	from: string;
	limit?: ParameterIndex;
	offset?: ParameterIndex;
	order?: AbstractSqlQueryOrderNode[];
	where?: AbstractSqlQueryWhereNode;
	intersect?: AbstractSqlQuery;
	parameters: (string | boolean | number)[];
}

export type AbstractSqlQueryOrderNode = {
	orderBy: AbstractQueryNodeSortTargets;
	direction: 'ASC' | 'DESC';
};

/**
 * An abstract WHERE clause.
 */
export interface AbstractSqlQueryWhereNode {
	type: 'condition';

	/* indicated of the condition should be negated using NOT */
	negation: boolean;

	/* value which will be compared to another value or expression. Functions will be supported soon. */
	target: SqlStatementColumn;

	/* an abstract comparator */
	operation: 'eq' | 'lt' | 'lte' | 'gt' | 'gte' | 'in' | 'between' | 'contains' | 'starts_with' | 'ends_with';

	/* a value to which the target will be compared */
	compareTo: CompareValueNode | CompareSetNode;
}

export interface CompareValueNode {
	type: 'value';
	parameterIndexes: number[];
}

export interface CompareSetNode {
	type: 'set';
	additionalOperator?: 'all' | 'some' | 'none';
	subQuery: AbstractSqlQuery;
}
/**
 * An actual vendor specific SQL statement with its parameters.
 * @example
 * ```
 * {
 * 		statement: 'SELECT * FROM "articles" WHERE "articles"."id" = $1;',
 * 		values: [99],
 * }
 * ```
 */
export interface ParameterizedSQLStatement {
	statement: string;
	parameters: (string | number | boolean)[];
}
