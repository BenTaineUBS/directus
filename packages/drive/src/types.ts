import { LocalFileSystemStorageConfig } from './LocalFileSystemStorage';

export type { LocalFileSystemStorageConfig };

export interface Storage {
	copy(src: string, dest: string): Promise<Response>;
	delete(location: string): Promise<DeleteResponse>;
	get(location: string, encoding?: string): Promise<ContentResponse<string>>;
	getBuffer(location: string): Promise<ContentResponse<Buffer>>;
	getStream(location: string, range?: Range): NodeJS.ReadableStream;
	getStat(_location: string): Promise<StatResponse>;
	move(_src: string, _dest: string): Promise<Response>;
	put(_location: string, _content: Buffer | NodeJS.ReadableStream | string, _type?: string): Promise<Response>;
	flatList(_prefix?: string): AsyncIterable<FileListResponse>;
}

export type StorageManagerSingleDiskConfig =
	| {
			driver: 'local';
			config: LocalFileSystemStorageConfig;
	  }
	| {
			driver: string;
			config: unknown;
	  };

export interface StorageManagerDiskConfig {
	[key: string]: StorageManagerSingleDiskConfig;
}

export interface StorageManagerConfig {
	/**
	 * The default disk returned by `disk()`.
	 */
	default?: string;
	disks?: StorageManagerDiskConfig;
}

export interface Response {
	raw: unknown;
}

export interface ExistsResponse extends Response {
	exists: boolean;
}

export interface ContentResponse<ContentType> extends Response {
	content: ContentType;
}

export interface SignedUrlOptions {
	/**
	 * Expiration time of the URL.
	 * It should be a number of seconds from now.
	 * @default `900` (15 minutes)
	 */
	expiry?: number;
}

export interface SignedUrlResponse extends Response {
	signedUrl: string;
}

export interface StatResponse extends Response {
	size: number;
	modified: Date;
}

export interface FileListResponse extends Response {
	path: string;
}

export interface DeleteResponse extends Response {
	wasDeleted: boolean | null;
}

export interface Range {
	start: number | undefined;
	end: number | undefined;
}
