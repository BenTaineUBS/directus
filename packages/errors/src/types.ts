import { ErrorCode } from './codes.js';
import type { ContainsNullValuesErrorExtensions } from './errors/contains-null-values.js';
import type { HitRateLimitErrorExtensions } from './errors/hit-rate-limit.js';
import type { IllegalAssetTransformationErrorExtensions } from './errors/illegal-asset-transformation.js';
import type { InvalidForeignKeyErrorExtensions } from './errors/invalid-foreign-key.js';
import type { InvalidPayloadErrorExtensions } from './errors/invalid-payload.js';
import type { InvalidProviderConfigErrorExtensions } from './errors/invalid-provider-config.js';
import type { InvalidQueryErrorExtensions } from './errors/invalid-query.js';
import type { MethodNotAllowedErrorExtensions } from './errors/method-not-allowed.js';
import type { NotNullViolationErrorExtensions } from './errors/not-null-violation.js';
import type { RangeNotSatisfiableErrorExtensions } from './errors/range-not-satisfiable.js';
import type { RecordNotUniqueErrorExtensions } from './errors/record-not-unique.js';
import type { RouteNotFoundErrorExtensions } from './errors/route-not-found.js';
import type { ServiceUnavailableErrorExtensions } from './errors/service-unavailable.js';
import type { UnprocessableContentErrorExtensions } from './errors/unprocessable-content.js';
import type { UnsupportedMediaTypeErrorExtensions } from './errors/unsupported-media-type.js';
import type { ValueOutOfRangeErrorExtensions } from './errors/value-out-of-range.js';
import type { ValueTooLongErrorExtensions } from './errors/value-too-long.js';

type Map = {
	[ErrorCode.ContainsNullValues]: ContainsNullValuesErrorExtensions;
	[ErrorCode.ContentTooLarge]: never;
	[ErrorCode.Forbidden]: never;
	[ErrorCode.IllegalAssetTransformation]: IllegalAssetTransformationErrorExtensions;
	[ErrorCode.InvalidCredentials]: never;
	[ErrorCode.InvalidForeignKey]: InvalidForeignKeyErrorExtensions;
	[ErrorCode.InvalidIp]: never;
	[ErrorCode.InvalidOtp]: never;
	[ErrorCode.InvalidPayload]: InvalidPayloadErrorExtensions;
	[ErrorCode.InvalidProvider]: never;
	[ErrorCode.InvalidProviderConfig]: InvalidProviderConfigErrorExtensions;
	[ErrorCode.InvalidQuery]: InvalidQueryErrorExtensions;
	[ErrorCode.InvalidToken]: never;
	[ErrorCode.LimitExceeded]: never;
	[ErrorCode.MethodNotAllowed]: MethodNotAllowedErrorExtensions;
	[ErrorCode.NotNullViolation]: NotNullViolationErrorExtensions;
	[ErrorCode.OutOfDate]: never;
	[ErrorCode.RangeNotSatisfiable]: RangeNotSatisfiableErrorExtensions;
	[ErrorCode.RecordNotUnique]: RecordNotUniqueErrorExtensions;
	[ErrorCode.RequestsExceeded]: HitRateLimitErrorExtensions;
	[ErrorCode.RouteNotFound]: RouteNotFoundErrorExtensions;
	[ErrorCode.ServiceUnavailable]: ServiceUnavailableErrorExtensions;
	[ErrorCode.TokenExpired]: never;
	[ErrorCode.UnexpectedResponse]: never;
	[ErrorCode.UnprocessableContent]: UnprocessableContentErrorExtensions;
	[ErrorCode.UnsupportedMediaType]: UnsupportedMediaTypeErrorExtensions;
	[ErrorCode.UserSuspended]: never;
	[ErrorCode.ValueOutOfRange]: ValueOutOfRangeErrorExtensions;
	[ErrorCode.ValueTooLong]: ValueTooLongErrorExtensions;
};

/** Verify all error codes are covered in the map. */
type Extends<T, U extends T> = U;

/** Map error codes to error extensions. */
export type ExtensionsMap = Extends<Record<(typeof ErrorCode)[keyof typeof ErrorCode], any>, Map>;
