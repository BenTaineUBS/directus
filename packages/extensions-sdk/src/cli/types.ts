import { LANGUAGES, LANGUAGES_SHORT } from './constants';

export type Language = typeof LANGUAGES[number];
export type LanguageShort = typeof LANGUAGES_SHORT[number];
