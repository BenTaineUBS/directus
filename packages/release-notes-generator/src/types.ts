import { getInfo } from '@changesets/get-github-info';
import type { NewChangesetWithCommit } from '@changesets/types';
import { TYPE_MAP } from './constants';

type AsyncReturnType<T extends (...args: any) => Promise<any>> = T extends (...args: any) => Promise<infer R> ? R : any;

type GithubInfo = AsyncReturnType<typeof getInfo>;

export type ChangesetsWithoutId = Map<string, Omit<NewChangesetWithCommit, 'id'>>;

export type Change = { summary: string; commit: string | undefined; githubInfo: GithubInfo | undefined };

export type Package = {
	name: string;
	changes: Change[];
};

type TypeValues = (typeof TYPE_MAP)[keyof typeof TYPE_MAP];
export type Type = { title: TypeValues; packages: Package[] };

export type PackageVersion = {
	name: string;
	version: string;
};
