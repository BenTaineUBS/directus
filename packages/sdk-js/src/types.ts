import { AxiosRequestConfig } from 'axios';
import { AuthOptions } from './handlers';

export type Item = Record<string, any>;
export type Payload = Record<string, any>;
export type PrimaryKey = string | number;

export enum Meta {
	TOTAL_COUNT = 'total_count',
	FILTER_COUNT = 'filter_count',
}

export type APIError = {
	message: string;
	extensions: {
		code: string;
	};
};

export type Response<T> = {
	data: T | null;
	meta?: Record<Meta, number>;
	errors?: APIError[];
};

export type Query = {
	fields?: string | string[];
	sort?: string;
	filter?: Filter;
	limit?: number;
	offset?: number;
	page?: number;
	single?: boolean;
	meta?: Meta[];
	search?: string;
	export?: 'json' | 'csv';
	deep?: Record<string, Query>;
};

export type Filter = {
	[keyOrOperator: string]: Filter | string | boolean | number | string[];
};

export type FilterOperator =
	| '_eq'
	| '_neq'
	| '_contains'
	| '_ncontains'
	| '_in'
	| '_nin'
	| '_gt'
	| '_gte'
	| '_lt'
	| '_lte'
	| '_null'
	| '_nnull'
	| '_empty'
	| '_nempty';

export type AuthStorage = {
	getItem: (key: string) => Promise<any>;
	setItem: (key: string, value: any) => Promise<any>;
};

export type DirectusOptions = {
	auth?: Partial<AuthOptions>;
	axiosConfig?: AxiosRequestConfig;
};
