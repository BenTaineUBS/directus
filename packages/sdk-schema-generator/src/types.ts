import type { Collection, Field, Relation } from '@directus/types';

export interface CliOptions {
	host: string;
	accessToken: string;
	file: string;
	naming: 'database' | 'camelcase' | 'pascalcase';
	rootName: string;
}

export interface BuilderOptions {
	nameTransform: string;
}

export interface RenderOptions {
	rootName: string;
	// nameTransform: string;
	indent: {
		amount: number;
		char: string;
	};
}

export interface DataModel {
	collections: Collection[];
	fields: Field[];
	relations: Relation[];
}

export type SchemaDefinition = Map<string, CollectionDefinition>;

export interface CollectionDefinition {
	name: string;
	system: boolean;
	singleton: boolean;
	fields: FieldDefinition[];
}

export interface FieldDefinition {
	name: string;
	type: MaybeArray<string>;
	nullable: boolean;
	primary_key: boolean;
	relation: MaybeArray<RelationDefinition> | null;
}

export interface RelationDefinition {
	collection: string;
	mutliple: boolean;
}

export type StringTransformer = (str: string) => string;

export type MaybeArray<T> = T | T[];

export type Nullable<T> = T | null | undefined;
