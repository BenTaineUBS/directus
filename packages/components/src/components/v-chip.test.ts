import { test, expect } from 'vitest';
import { mount } from '@vue/test-utils';

import VChip from './v-chip.vue';
import { GlobalMountOptions } from '@vue/test-utils/dist/types';

const global: GlobalMountOptions = {
	stubs: ['v-icon'],
};

test('Mount component', () => {
	expect(VChip).toBeTruthy();

	const wrapper = mount(VChip, {
		global,
	});

	expect(wrapper.html()).toMatchSnapshot();
});

test('modelValue prop', async () => {
	const wrapper = mount(VChip, {
		props: {
			modelValue: 'my value',
		},
		global,
	});

	expect(wrapper.get('input').element.value).toBe('my value');

	await wrapper.find('input').setValue('my value1');

	expect(wrapper.emitted()['update:modelValue'][0]).toEqual(['my value1']);
});
