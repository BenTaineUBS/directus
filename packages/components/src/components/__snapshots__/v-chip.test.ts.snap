// Vitest Snapshot v1

exports[`Mount component 1`] = `"<!--v-if-->"`;
